core/src/test/java/org/elasticsearch/search/aggregations/metrics/stats/multifield/MultiFieldStatsIT.java

```diff
@@ -0,0 +1,380 @@
+import org.elasticsearch.search.aggregations.metrics.AbstractNumericTestCase;
+import org.elasticsearch.search.aggregations.metrics.AvgIT;
+import org.elasticsearch.search.aggregations.metrics.ValueCountIT;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
+import static org.elasticsearch.index.query.QueryBuilders.matchAllQuery;
+import static org.elasticsearch.search.aggregations.AggregationBuilders.global;
+import static org.elasticsearch.search.aggregations.AggregationBuilders.histogram;
+import static org.elasticsearch.search.aggregations.AggregationBuilders.multifieldStats;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertSearchResponse;
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+/**
+ * MultiFieldStats aggregation integration test
+ */
+public class MultiFieldStatsIT extends AbstractNumericTestCase {
+    protected static String[] valFieldNames = new String[] {"value", "randVal1", "randVal2", "values"};
+    protected static String[] singleValFields = Arrays.copyOf(valFieldNames, 3);
+    protected static MultiFieldStatsResults multiFieldStatsResults = new MultiFieldStatsResults();
+    protected static long numDocs = 10000L;
+    protected static final double TOLERANCE = 1e-6;
+    protected static final String aggName = "multifieldstats";
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Arrays.asList(
+            AvgIT.ExtractFieldScriptPlugin.class,
+            ValueCountIT.FieldValueScriptPlugin.class);
+    }
+
+    @Override
+    public void setupSuiteScopeCluster() throws Exception {
+        super.setupSuiteScopeCluster();
+        createIndex("mfs_idx");
+        List<IndexRequestBuilder> builders = new ArrayList<>();
+
+        Map<String, Double> vals = new HashMap<>(3);
+        for (int i = 0; i < numDocs; i++) {
+            vals.put(valFieldNames[0], (double)i+1);
+            vals.put(valFieldNames[1], randomDouble());
+            vals.put(valFieldNames[2], randomDouble());
+            vals.put(valFieldNames[3], ((i+2) + (i+3)) / 2.0);
+            builders.add(client().prepareIndex("idx", "type", ""+i).setSource(jsonBuilder()
+                .startObject()
+                .field(valFieldNames[0], vals.get(valFieldNames[0]))
+                .field(valFieldNames[1], vals.get(valFieldNames[1]))
+                .field(valFieldNames[2], vals.get(valFieldNames[2]))
+                .startArray(valFieldNames[3]).value(i+2).value(i+3).endArray()
```

Comment:
hmmm... I can add random field omission? Since these are numeric fields an IAE is thrown if `null` or `NaN` is passed.